### PR TITLE
Set resolver focus

### PR DESCRIPTION
**Pull request by mlandres**
[bsc#1146415](https://bugzilla.suse.com/show_bug.cgi?id=1146415), continues openSUSE/libzypp#185.
Add '--solver-focus' option and docs.

### CODE REVIEW

File: src/commands/solveroptionset.cc [hunk 1]
```diff
@@ -7,6 +7,28 @@
...
 
+namespace
+{
+  // Map string argument to ResolverFocus
+  ZyppFlags::Value ResolverFocusArgType( ResolverFocus & optval_r )
+  {
+    return ZyppFlags::Value (
+      // DefValueFun
+      []() -> boost::optional<std::string>
+      { return boost::optional<std::string>(); },
```
Comment (bzeller) [inline src/commands/solveroptionset.cc]: You can use ZyppFlags::noDefaultValue for that

File: doc/zypper.8.d/option_Solver_Flags_Common.txt [hunk 2]
```diff
@@ -6,3 +6,6 @@
 
 *-R*, *--no-force-resolution*::
 	Do not force the solver to find a solution. Instead, report dependency problems and prompt the user to resolve them manually. This is the default except when removing packages (*zypper remove*).
+
+*--solver-focus* 'MODE'::
+	Set the solvers general attitude when resolving a job. Valid modes are  *Job*, *Installed* or *Update*. See section *Package Dependencies* for details.
```
Comment (bzeller) [inline doc/zypper.8.d/option_Solver_Flags_Common.txt]: Double space between are and *Job*

File: doc/zypper.8.txt [hunk 3]
```diff
@@ -102,7 +102,16 @@ Throughout this manual we will often refer to resolvables simply as 'packages' a
 
 Package Dependencies
 ~~~~~~~~~~~~~~~~~~~~
-Software packages depend on each other in various ways. Packages usually 'require' or 'recommend' other packages, but they can also 'conflict' with other packages. Packages may support specific hardware or language settings. Zypper uses a 'dependency solver' to find out which packages need to be installed to satisfy the user's request. See https://en.opensuse.org/Libzypp/Dependencies for more information.
+Software packages depend on each other in various ways. Packages usually 'require' or 'recommend' other packages, but they can also 'conflict' with other packages. Packages may support specific hardware or language settings. Zypper uses a 'dependency solver' to find out which packages need to be installed to satisfy the user's request.
```
Comment (bzeller) [inline doc/zypper.8.txt]: `Packages usually 'require' or 'recommend' other packages, but they can also 'conflict' with other packages.`
->
`Packages usually 'require' or 'recommend' other packages, but they can also 'conflict' with them.`

Comment (bzeller) [reply doc/zypper.8.txt]: Repeating the `other packages` part seems wrong to me but I'm no native speaker so not sure if that is the case.

File: doc/zypper.8.txt [hunk 4]
```diff
@@ -102,7 +102,16 @@ Throughout this manual we will often refer to resolvables simply as 'packages' a
...
+
+If you do not request a specific version of a package the solver will pick a reasonable one. The solvers general attitude when resolving a job is to focus on installing the best version of the requested package and to add or update dependencies as they are needed. Aside from this 'Focus on Job', which is the default, two other focus modes are available:
+
+In 'Focus on Installed' mode the solver focuses on applying as little changes to the installed packages as needed. Choosing an older version of a requested  package is valid if it's dependencies require less changes to the system. The solver will try to avoid updating already installed packages.
```
Comment (bzeller) [inline doc/zypper.8.txt]: Double space between `requested` and `package`

File: src/commands/solveroptionset.cc [hunk 5]
```diff
@@ -40,7 +62,8 @@ std::vector<zypp::ZyppFlags::CommandGroup> SolverCommonOptionSet::options()
       {
         { "debug-solver", '\0', ZyppFlags::NoArgument, ZyppFlags::TriBoolType( set._debugSolver, ZyppFlags::StoreTrue ), _("Create a solver test case for debugging.") },
         { "force-resolution", '\0', ZyppFlags::NoArgument, ZyppFlags::TriBoolType( set._forceResolution, ZyppFlags::StoreTrue ), _("Force the solver to find a solution (even an aggressive one) rather than asking.") },
-        { "no-force-resolution", 'R', ZyppFlags::NoArgument, ZyppFlags::TriBoolType( set._forceResolution, ZyppFlags::StoreFalse ), _("Do not force the solver to find solution, let it ask.") }
+        { "no-force-resolution", 'R', ZyppFlags::NoArgument, ZyppFlags::TriBoolType( set._forceResolution, ZyppFlags::StoreFalse ), _("Do not force the solver to find solution, let it ask.") },
```
Comment (bzeller) [inline src/commands/solveroptionset.cc]: `Do not force the solver to find solution` -> Do not force the solver to find **a** solution.
Not a new line but while at it we can fix the missing word :)

File: doc/zypper.8.txt (hunk 3 above)
Comment (bzeller) [reply doc/zypper.8.txt]: `with those` might also work...

File: src/commands/solveroptionset.cc (hunk 5 above)
Comment (mlandres) [reply src/commands/solveroptionset.cc]: Will do in a separate commit directly in master. It needs to adjust the all .po files in order not to lose the translations.

File: doc/zypper.8.txt (hunk 4 above)
Comment (mlandres) [reply doc/zypper.8.txt]: done

File: doc/zypper.8.txt (hunk 3 above)
Comment (mlandres) [reply doc/zypper.8.txt]: 'with them' sounds right to me.

File: doc/zypper.8.d/option_Solver_Flags_Common.txt (hunk 2 above)
Comment (mlandres) [reply doc/zypper.8.d/option_Solver_Flags_Common.txt]: done

**Pull request merged**